packages/server/test/node.spec.ts

```diff
@@ -53,7 +53,7 @@ describe('Node Specific Cases', () => {
         });
       });
       const serverAdapter = createServerAdapter<{
-        req: IncomingMessage;
+        req: NodeRequest;
```

Comment:
Why do we need this change?

Reply:
IncomingMessage is bubbling up in  normalizeNodeRequest in `const rawRequest = nodeRequest.raw || nodeRequest.req || nodeRequest;` and passed down into buildFullUrl() => getHostnameWithPort().

Reply:
the main issue is here.